registrasion/reporting/views.py

```diff
@@ -240,6 +241,55 @@ def group_by_cart_status(queryset, order, values):
     return values
 
 
+@report_view("Limits")
+def limits(request, form):
+    ''' Shows the summary of sales against stock limits. '''
+
+    line_items = commerce.LineItem.objects.filter(
+        invoice__status=commerce.Invoice.STATUS_PAID,
+    ).annotate(
+        total_quantity=Sum("quantity"),
+    )
+
+    quantities = {}
+    for line_item in line_items.all():
+        quantities[line_item.product.name] = line_item.quantity
+
+    print 'quantities', quantities
+
+    limits = conditions.TimeOrStockLimitFlag.objects.all().order_by("-limit")
+
+    headings = ["Product", "Quantity"]
+
+    reports = []
+    for limit in limits:
+        data = []
+        total = 0
+        for product in limit.products.all():
+            if product.name in quantities:
+                total += quantities[product.name]
+                data.append([product.name, quantities[product.name]])
+        if limit.limit:
+            data.append(['(TOTAL)', '%s/%s' % (total, limit.limit)])
+        else:
+            data.append(['(TOTAL)', total])
+
+        description = limit.description
+        extras = []
+        if limit.start_time:
+            extras.append('Starts: %s' % (limit.start_time))
```

Comment:
As a matter of practice, I try to avoid doing formattish things like this on the server side.   Putting things like "Starts: ..." in the template is preferred.  (Sorry to nitpick.)